invokeai/app/util/step_callback.py

```diff
@@ -56,23 +57,51 @@ def stable_diffusion_step_callback(
 
     # TODO: only output a preview image when requested
 
-    # origingally adapted from code by @erucipe and @keturn here:
-    # https://discuss.huggingface.co/t/decoding-latents-to-rgb-without-upscaling/23204/7
-
-    # these updated numbers for v1.5 are from @torridgristle
-    v1_5_latent_rgb_factors = torch.tensor(
-        [
-            #    R        G        B
-            [0.3444, 0.1385, 0.0670],  # L1
-            [0.1247, 0.4027, 0.1494],  # L2
-            [-0.3192, 0.2513, 0.2103],  # L3
-            [-0.1307, -0.1874, -0.7445],  # L4
-        ],
-        dtype=sample.dtype,
-        device=sample.device,
-    )
-
-    image = sample_to_lowres_estimated_image(sample, v1_5_latent_rgb_factors)
+    if base_model in [BaseModelType.StableDiffusionXL, BaseModelType.StableDiffusionXLRefiner]:
+        sdxl_latent_rgb_factors = torch.tensor(
+            [
+                #   R        G        B
+                [0.3816, 0.4930, 0.5320],
+                [-0.3753, 0.1631, 0.1739],
+                [0.1770, 0.3588, -0.2048],
+                [-0.4350, -0.2644, -0.4289],
+            ],
+            dtype=sample.dtype,
+            device=sample.device,
+        )
+
+        sdxl_smooth_matrix = torch.tensor(
+            [
+                # [ 0.0478,  0.1285,  0.0478],
+                # [ 0.1285,  0.2948,  0.1285],
+                # [ 0.0478,  0.1285,  0.0478],
```

Comment:
Is this the matrix from another project? Maybe best to remove depending on license

Reply:
Also give yourself credit for experimentally determining your own!

Reply:
Ok, i'll add my credits)
This just filter matrix before I reduced it's strength.
So, one that bellow based on this(multiplicated by 0.75 and recalculated center value)